File: sdk/node/hwid.js
const { execSync } = require('child_process');
const crypto = require('crypto');

function getWmiProperty(command) {
    try {
        const output = execSync(command).toString().trim();
        const lines = output.split('\n').map(l => l.trim()).filter(l => l.length > 0);
        if (lines.length > 1) {
            return lines[1];
        }
        return "unknown";
    } catch (e) {
        return "unknown";
    }
}

function getHardwareSignals() {
    if (process.platform !== 'win32') {
        return { platform: process.platform, arch: process.arch };
    }
    return {
        cpuId: getWmiProperty('wmic cpu get processorid'),
        motherboard: getWmiProperty('wmic baseboard get serialnumber'),
        uuid: getWmiProperty('wmic csproduct get uuid'),
        disk: getWmiProperty('wmic diskdrive get serialnumber')
    };
}

function getHwid() {
    const signals = getHardwareSignals();
    const baseStr = `${signals.uuid}|${signals.motherboard}|${signals.cpuId}`;
    return crypto.createHash('sha256').update(baseStr).digest('hex');
}

module.exports = { getHwid, getHardwareSignals };

File: sdk/python/hwid.py
import subprocess
import hashlib
import platform

def get_wmi_property(cmd):
    try:
        output = subprocess.check_output(cmd, shell=True).decode().strip()
        # Clean up output (wmic often adds headers and extra lines)
        lines = [line.strip() for line in output.split('\n') if line.strip()]
        if len(lines) > 1:
            return lines[1]
        return "unknown"
    except:
        return "unknown"

def get_hardware_signals():
    return {
        "cpuId": get_wmi_property("wmic cpu get processorid"),
        "motherboard": get_wmi_property("wmic baseboard get serialnumber"),
        "uuid": get_wmi_property("wmic csproduct get uuid"),
        "disk": get_wmi_property("wmic diskdrive get serialnumber")
    }

def get_hwid():
    signals = get_hardware_signals()
    base_str = f"{signals['uuid']}|{signals['motherboard']}|{signals['cpuId']}"
    return hashlib.sha256(base_str.encode()).hexdigest()
